File: scripts/test_db.js
// scripts/test_db.ts
import "dotenv/config"; // Load environment variables
import { Pool } from "pg";

const pool = new Pool({
  user: process.env.DB_USER,
  host: process.env.DB_HOST,
  database: process.env.DB_NAME,
  password: process.env.DB_PASSWORD,
  port: parseInt(process.env.DB_PORT || "5432", 10), // Default to 5432 if not specified
});

async function testDatabase() {
  try {
    const result = await pool.query("SELECT NOW()");
    console.log("Connection test successful:", result.rows[0].now);
  } catch (error) {
    console.error("Error testing database:", error);
  } finally {
    await pool.end();
  }
}

testDatabase();

File: scripts/python/test_db.py
import psycopg2

# Database connection parameters
DB_HOST = "localhost"
DB_NAME = "the_box_2025"
DB_USER = "test_user"
DB_PASSWORD = "password" # Replace with your PostgreSQL password


conn = psycopg2.connect(host=DB_HOST, database=DB_NAME, user=DB_USER, password=DB_PASSWORD)
cur = conn.cursor()
cur.execute("SELECT version();")
version = cur.fetchone()[0]
print(version)
cur.close()
conn.close()